### PR TITLE
Added: Gluon recommender system notebook

### PR DESCRIPTION
**Pull request by djarpin**

### CODE REVIEW

**Comment by orchidmajumder** [comment]
I can review notebooks written in Gluon. I'll take this review on myself.

File: introduction_to_applying_machine_learning/gluon_recommender_system/recommender.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        self.dropout_p = dropout_p
+        self.num_emb = num_emb
+        
+        with self.name_scope():
+            self.user_embeddings = gluon.nn.Embedding(max_users, num_emb)
+            self.item_embeddings = gluon.nn.Embedding(max_items, num_emb)
+            self.dropout = gluon.nn.Dropout(dropout_p)
+            self.dense = gluon.nn.Dense(num_emb, activation='relu')
+            
+    def forward(self, users, items):
```
Comment (orchidmajumder) [inline introduction_to_applying_machine_learning/gluon_recommender_system/recommender.py]: Please use hybrid_forward. The operation here is simple and should be easily hybridized.

File: introduction_to_applying_machine_learning/gluon_recommender_system/recommender.py [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+        print("epoch: {}".format(e))
+        for i, (user, item, label) in enumerate(train_iter):
+            try:
+                user = user.as_in_context(ctx).reshape((batch_size,))
+                item = item.as_in_context(ctx).reshape((batch_size,))
+                label = label.as_in_context(ctx).reshape((batch_size,))
+                with mx.autograd.record():
+                    output = net(user, item)               
+                    loss = loss_function(output, label)
+                    loss.backward()
```
Comment (orchidmajumder) [inline introduction_to_applying_machine_learning/gluon_recommender_system/recommender.py]: nit: loss.backward() does not need to be within autograd block.

File: introduction_to_applying_machine_learning/gluon_recommender_system/recommender.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                net.collect_params().values()
```
Comment (orchidmajumder) [inline introduction_to_applying_machine_learning/gluon_recommender_system/recommender.py]: This line seems redundant.

**Comment by djarpin** [comment]
Thanks @orchidmajumder !  Updates and fixes made.
